src/nupic/bindings/sparse_matrix.i

```diff
@@ -1002,6 +1002,27 @@ def __div__(self, other):
   }
 
 
+  %pythoncode %{
+    def nNonZerosPerRowOnCols(self, rows, cols):
+      rows = numpy.asarray(rows, dtype="uint32")
```

Comment:
`np.asarray` also converts lists and tuples to numpy arrays. We don't have any C code that does that.

Also I think `NumpyVectorWeakRefT` is our only class that doesn't copy the array. `NumpyVectorT` always copies the array, even if it's the correct dtype.